### PR TITLE
Add NodeJS 12 image based on UBI8

### PR DESCRIPTION
**Pull request by michaelsauter**

### CODE REVIEW

**Review by faust2199 — COMMENTED** [review]
You might need to update .github\workflows\continuous-integration-workflow.yml so that the renamed Dockerfile can be picked up.

**Comment by michaelsauter** [comment]
@faust2199 Updated!

File: .github/workflows/continuous-integration-workflow.yml [hunk 1]
```diff
@@ -55,7 +55,23 @@ jobs:
         name: Build docker image
         working-directory: common/jenkins-agents/nodejs12/docker
         run: |
-            docker build --tag agent-nodejs12-test \
+            docker build --tag agent-nodejs12-test --file Dockerfile.rhel7 \
```
Comment (faust2199) [inline .github/workflows/continuous-integration-workflow.yml:58]: I don't know if it matters, but we used to add a suffix to the tag with each version for the go agent.

File: .github/workflows/continuous-integration-workflow.yml [hunk 2]
```diff
@@ -55,7 +55,23 @@ jobs:
...
+    runs-on: ubuntu-18.04
+    steps:
+      -
+        name: Checkout repository
+        uses: actions/checkout@v2.0.0
+      -
+        name: Build docker image
+        working-directory: common/jenkins-agents/nodejs12/docker
+        run: |
+            docker build --tag agent-nodejs12-test --file Dockerfile.ubi8 \
```
Comment (faust2199) [inline .github/workflows/continuous-integration-workflow.yml:74]: See comment above.

File: common/jenkins-agents/nodejs12/docker/Dockerfile.ubi8 [hunk 3]
```diff
@@ -0,0 +1,61 @@
+FROM opendevstackorg/ods-jenkins-agent-base-ubi8:latest
+
+# Labels consumed by Red Hat build service
+LABEL com.redhat.component="jenkins-agent-nodejs-12-rhel7-container" \
+      name="openshift4/jenkins-agent-nodejs-12-rhel7" \
+      architecture="x86_64" \
+      io.k8s.display-name="Jenkins Agent Nodejs" \
+      io.k8s.description="The jenkins agent nodejs image has the nodejs tools on top of the jenkins slave base image." \
+      io.openshift.tags="openshift,jenkins,agent,nodejs" \
+      maintainer="openshift-dev-services+jenkins@redhat.com"
```
Comment (faust2199) [inline common/jenkins-agents/nodejs12/docker/Dockerfile.ubi8:10]: Are those lines still useful? If so, we might want to update them to reflect the fact rhel7 is no longer used here.

File: .github/workflows/continuous-integration-workflow.yml (hunk 1 above)
Comment (michaelsauter) [reply .github/workflows/continuous-integration-workflow.yml]: Changed in 4aeeeb6.

File: .github/workflows/continuous-integration-workflow.yml (hunk 2 above)
Comment (michaelsauter) [reply .github/workflows/continuous-integration-workflow.yml]: Changed in 4aeeeb6.

File: common/jenkins-agents/nodejs12/docker/Dockerfile.ubi8 (hunk 3 above)
Comment (michaelsauter) [reply common/jenkins-agents/nodejs12/docker/Dockerfile.ubi8:10]: Those are copied from the upstream image. They look outdated to me as well, however let's address labelling in another issue, and then make it consistent across all images.

**Review by faust2199 — APPROVED** [review]

**Pull request merged**